.gitmodules

```diff
@@ -1,3 +1,7 @@
 [submodule "include/cxxopts"]
 	path = include/cxxopts
 	url = https://github.com/jarro2783/cxxopts
+[submodule "include/polylineencoder"]
+	path = include/polylineencoder
+	url = https://github.com/vahancho/polylineencoder
+	branch = v1.1.2
```

Comment:
I'm not familiar with the submodule options in git and a bit puzzled since there is no `v1.1.2` branch in `polylineencoder`, just a tag. Does specifying this will fallback to using the tag if no branch exists?

Reply:
I have to say this sounds a bit confusing to me too. When you want to lock submodule to a spesific tag this trick becomes necessary AFAIK. Because when you run something like this.

```
git submodule update --init --remote
```

If branch isn't defined, submodule upgraded to latest commit on default branch and pollute tag-lock. When there is no release branch on the target repo, it isn't possible to determine the correct branch name. 

Since we have already locked the submodule to a tag, I chose to use the branch of the submodule to show its version for informational purposes.

This will only prevent you from updating submodules as above. When a submodule needs to be updated, it will be enough to run the following. Actually otherwise you should consider to not polluting submodule. 

```
git submodule set-branch --branch {desired_version} include/{module_name}
cd include/{module_name}
git fetch && git checkout {desired_version}
```

I don't think what I mentioned above is best-practice. Do you have any suggestions?

Reply:
> Since we have already locked the submodule to a tag, I chose to use the branch of the submodule to show its version for informational purposes.

Right, I think I get it now. The actual commit hash we point to is not relying on this value but stored by git anyway and shows up on status:

```
$ git submodule status
 c74846a891b3cc3bfa992d588b1295f528d43039 include/cxxopts (v3.0.0)
 bea1070d8dd974d273f39f1cf2c4b393f5594f99 include/polylineencoder (1.0.0-29-gbea1070)
```

I think we can remove the `branch`  value from `.gitmodules` since this is somehow redundant, and not pointing to an actual branch anyway.

Reply:
That makes sense 👍